helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -317,12 +317,18 @@ private void reportTopStateHandoffFailIfNecessary(ResourceControllerDataProvider
     String partitionName = partition.getPartitionName();
     MissingTopStateRecord record = missingTopStateMap.get(resourceName).get(partitionName);
     long startTime = record.getStartTimeStamp();
-    if (startTime > 0 && System.currentTimeMillis() - startTime > durationThreshold && !record
-        .isFailed()) {
+    long missingDuration = System.currentTimeMillis() - startTime;
+    if (startTime > 0 && missingDuration > durationThreshold && !record.isFailed()) {
       record.setFailed();
       missingTopStateMap.get(resourceName).put(partitionName, record);
+      // Since top state handoff has not completed yet we can't log helix top state latency but can log since how long
+      // top state is missing.
+      LogUtil.logInfo(LOG, _eventId, String.format(
```

Comment:
If i understand correctly, this is what ESPRESSO wants it.

Reply:
Synced up offline. Reverting other log statement and would because it has to be done with caution to make sure there is no log flooding. I will separate out that change in separate PR